From 9cf1e56a00788e7c853772acc62803d8d794b667 Mon Sep 17 00:00:00 2001
From: Brahma Reddy Battula <brahma@apache.org>
Date: Fri, 21 Jul 2017 15:13:20 +0200
Subject: [PATCH 2680/2848] HDFS-12177. NameNode exits due to setting
 BlockPlacementPolicy loglevel to Debug.
 Contributed by Jiandan Yang.

(cherry picked from commit f65dc6ee952d63b80766e1a3a298558443678ae9)

Change-Id: I603cb815acbd404f5e652af4111d57eacad98957
Ref: CDH-61971
---
 .../BlockPlacementPolicyDefault.java               |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockPlacementPolicyDefault.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockPlacementPolicyDefault.java
index cb9c67e..7c0cde6 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockPlacementPolicyDefault.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockPlacementPolicyDefault.java
@@ -696,7 +696,7 @@ protected DatanodeStorageInfo chooseRandom(int numOfReplicas,
       }
       Preconditions.checkState(excludedNodes.add(chosenNode), "chosenNode "
           + chosenNode + " is already in excludedNodes " + excludedNodes);
-      if (LOG.isDebugEnabled()) {
+      if (LOG.isDebugEnabled() && builder != null) {
         builder.append("\nNode ").append(NodeBase.getPath(chosenNode))
             .append(" [");
       }
@@ -725,7 +725,7 @@ protected DatanodeStorageInfo chooseRandom(int numOfReplicas,
           }
         }
 
-        if (LOG.isDebugEnabled()) {
+        if (LOG.isDebugEnabled() && builder != null) {
           builder.append("\n]");
         }
 
-- 
1.7.9.5

